src/service/url-replacements-impl.js

```diff
@@ -47,27 +63,32 @@ export class UrlReplacements {
     /** @private {!RegExp|undefined} */
     this.replacementExpr_ = undefined;
 
-    /** @private @const {!Object<string, function(*):*>} */
+    /** @private @const {!Object<string, !ReplacementDef>} */
     this.replacements_ = this.win_.Object.create(null);
 
-    /** @private @const {function():!Promise<?AccessService>} */
+    /** @private @const {function(Window):Promise<(AccessService|null)>} */
     this.getAccessService_ = accessServiceForOrNull;
 
     /** @private @const {!Promise<?Object<string, string>>} */
     this.variants_ = variantForOrNull(win);
 
     /** @private {boolean} */
     this.initialized_ = false;
+
+    // NOTE: getService prevents multiple instances of same service across
+    // binaries (e.g. extensions) therefore ensures only one click listener.
+    this.registerClickListener_();
```

Comment:
Having both phases is OK. The code over there already handles the case
where the target is a child of the link tag.

On Fri, Jul 22, 2016 at 12:12 PM, keithwrightbos <notifications@github.com>
wrote:

> In src/service/url-replacements-impl.js
> <https://github.com/ampproject/amphtml/pull/4153#discussion_r71930594>:
>
> >      this.getAccessService_ = accessServiceForOrNull;
> >
> >      /** @private @const {!Promise<?Object<string, string>>} */
> >      this.variants_ = variantForOrNull(win);
> >
> >      /** @private {boolean} */
> >      this.initialized_ = false;
> > +
> > +    // NOTE: getService prevents multiple instances of same service across
> > +    // binaries (e.g. extensions) therefore ensures only one click listener.
> > +    this.registerClickListener_();
>
> @cramforce <https://github.com/cramforce> it will need to be an
> additional click listener if we want to capture phase. SG?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ampproject/amphtml/pull/4153/files/e15faa40318d78d021f6651f018c0b008eaababc#r71930594>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAFeT0tJ9C4cDyLEM4yIzvfMFjWQr7kfks5qYRYTgaJpZM4JR8if>
> .
>